package-tests/tests/package_tests.rs

```diff
@@ -105,8 +106,205 @@ fn test_package_input(file: &Path) -> datatest_stable::Result<()> {
+            .expect("Spawn `cargo run` process");
+
+        let result = String::from_utf8(cmd.wait_with_output().unwrap().stdout)
+            .expect("Expected result to be utf8");
+        serde_json::from_str(&result).expect("Not a valid package manifest")
+    };
+
+    let cargo_manifest: Value = {
+        let cmd = Command::new(env!("CARGO"))
+            .arg("read-manifest")
+            .arg(format!("--manifest-path={}", file.to_string_lossy()))
+            .stdout(Stdio::piped())
+            .spawn()
+            .expect("Spawn `cargo read-manifest` process");
+
+        let result = String::from_utf8(cmd.wait_with_output().unwrap().stdout)
+            .expect("Expected result to be utf8");
+        serde_json::from_str(&result).expect("Cargo read-manifest not valid JSON")
+    };
+
+    // Make sure to explicitly drop the lock, so that we don't drop it earlier implicitly. It is ok
+    // to drop it now since we are done with the package itself.
+    // Even though the tests aren't done, we are done with running Cargo and thus don't need
+    // exclusive access to the directory anymore.
+    drop(lock);
+
+    // First, we want to check the cargo manifest validity
+    // Get the last component of the path, which is the folder name
+    let package_folder = package_path.components().next_back().unwrap().as_os_str();
+    // Get the project name from cargo manifest
+    let project_name = cargo_manifest["name"].as_str().unwrap();
+
+    assert_eq!(
+        project_name, package_folder,
+        "Cargo project name should have the same name as the folder the package is in"
+    );
+
+    // Loop though all cargo targets and find all "bin" targets
+    let bin_targets: Vec<&serde_json::Map<String, Value>> = cargo_manifest["targets"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .filter_map(|target| {
+            let target = target.as_object().unwrap();
+            target["kind"]
+                .as_array()
+                .unwrap()
+                .iter()
+                .any(|t| t.as_str().unwrap() == "bin")
+                .then_some(target)
+        })
+        .collect();
+
+    assert_eq!(
+        bin_targets.len(),
+        1,
+        "Cargo project should have exactly one binary target"
+    );
+
+    assert_eq!(
+        bin_targets[0]["name"].as_str().unwrap(),
+        package_folder,
+        "Cargo project target should have the same name as the folder the package is in"
+    );
+
+    // Now, let's check the package manifest
+    // Check that the name is lowercase/digit/_/- and starts lowercase
+    let name = manifest.name;
+    let ok_name = name.starts_with(|c: char| c.is_ascii_lowercase())
+        && name
+            .chars()
+            .all(|c| c.is_ascii_lowercase() || c.is_ascii_digit() || c == '_' || c == '-');
+
+    assert!(ok_name, "Package name must start with lowercase letter, and must only contain lowercase letters, digits, underscores and dashes");
```

Comment:
Maybe it could also check that the name provided by the package manifest matches the others.